File: backend/Models/Notification.cs
namespace LocalHire.Api.Models;

public sealed class Notification
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public required string Type { get; set; }
    public required string Title { get; set; }
    public required string Message { get; set; }
    public string? Link { get; set; }
    public bool IsRead { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public DateTimeOffset? ReadAt { get; set; }

    public User User { get; set; } = null!;
}

File: integration-tests/features/db_models/notification.py
"""Row model for the ``Notifications`` table."""

from dataclasses import dataclass, field
from datetime import datetime
from typing import Optional

from features.db_models.model import Model
from features.utils.util import utc_now


@dataclass
class Notification(Model):
    id: str
    user_id: str
    type: str  # NewApplication | Shortlisted | Hired | Rejected | JobUpdated
    title: str
    message: str
    link: Optional[str] = None
    is_read: bool = False
    created_at: datetime = field(default_factory=utc_now)
    read_at: Optional[datetime] = None
